File: ui/src/l10n/tokens.ts
/** Strings that are NOT copy, and must never reach a translator.
 *
 *  The Flutter client keeps these in `l10n/tokens.dart` for the same reason
 *  (`docs/i18n.md`, rule 1): a naive extractor sweeps glyphs, shell commands
 *  and wire-format examples into the catalog, where a translator is then asked
 *  to localize `jeliyad` or `▦`. Worse, `docs/glossary-fr.md` Tier 2 makes some
 *  of these a correctness matter — a connection path badge must render exactly
 *  what the daemon reported, so "translating" `direct` would be a lie about the
 *  network.
 *
 *  Everything here is deliberately outside `Catalog`.
 */

/** Decorative glyphs. Every one of these is `aria-hidden` at its call site —
 *  they carry no information the adjacent label does not already give. */
export const Glyph = {
  rooms: '▦',
  fleet: '✦',
  settings: '⚙',
  hex: '⬡',
  copy: '⧉',
  file: '⎘',
  pipe: '⤳',
  send: '➤',
  close: '✕',
  chevronDown: '⌄',
  disclosureOpen: '▾',
  disclosureClosed: '▸',
  back: '‹',
  more: '⋮',
  pinOn: '★',
  pinOff: '☆',
  archive: '⇩',
  restore: '⇧',
  create: '⊕',
  join: '⇥',
  add: '＋',
  externalLink: '↗',
  openRoom: '⇱',
  previous: '←',
  share: '↗',
  verified: '✓',
  failed: '✕',
} as const;

/** Punctuation and layout separators. Not sentences — a list of independent
 *  facts may be joined with these, but a SENTENCE may not (rule 2). */
export const Punct = {
  /** The decorative separator between independent facts on a meta line. */
  metaSep: ' · ',
  emDash: '—',
  /** Shown where a value is genuinely absent, never as a zero. */
  missingValue: '—',
  /** Count cap on a tab badge. */
  countCap: '99+',
} as const;

/** The brand. Never translated, never re-derived — DESIGN.md makes the wordmark
 *  a single source of truth. */
export const BRAND = 'Jeliya';

/** Language endonyms for the picker. A language is named in ITSELF, so these
 *  are the one set of language names that must never be translated — and a
 *  picker showing a bare ISO code is a bug the locale test catches. */
export const LANGUAGE_NAMES: Record<string, string> = {
  en: 'English',
  fr: 'Français',
};

/** Wire-format examples used as input placeholders. These show the user the
 *  SHAPE of a machine value; translating them would teach the wrong shape. */
export const Example = {
  ticket: 'roomtkt1… or roomtkt1…#<endpoint_id>@host:port',
  peerAddress: '<endpoint_id>@203.0.113.7:4242',
  pipeTarget: '127.0.0.1:3000',
  expirySeconds: '3600',
  identityId: '64-hex identity id',
  roomName: 'Build Iroh Rooms MVP',
  alias: 'e.g. Maya R.',
  selfLabel: 'e.g. Alex',
} as const;

/** Shell commands and paths. Copy-pasteable machine input — a translated
 *  command does not run. */
export const Command = {
  daemon: 'jeliyad',
  daemonPortParam: '?daemon=<port>',
  npmInstall: 'npm install',
  daemonPath: 'JELIYAD="$(command -v jeliyad)"',
  agentGuide: 'docs/agent-guide.md',
} as const;

/** The issue tracker. A URL is not copy. */
export const ISSUE_URL = 'https://github.com/kortiene/jeliya/issues/new';

File: crates/jeliya-ui/src/l10n/tokens.rs
//! Never-translate constants (i18n.md rule 1; spec §5.1).
//!
//! Glyphs, the brand, shell/wire examples, and language endonyms are **outside**
//! the [`Catalog`](super::Catalog) by design: they are the same in every
//! language, so routing them through a translated message would invite a
//! translator to "fix" a value that must not change and would make the French
//! catalog claim to translate the brand. The literal-copy gate treats this
//! module as exempt for exactly that reason (the React `EXEMPT_FILES` records
//! the same decision for `ui/src/l10n/tokens.ts`).

/// The product name. Tier 3 of the never-translate glossary.
pub const BRAND: &str = "Jeliya";

/// The middle dot used between two facts in a status line (`client · Ready`).
/// A glyph, not copy — it carries no words to translate.
pub const MIDDLE_DOT: &str = "·";

/// The endonym of each supported language, for a (later) language picker. An
/// endonym is written in its own language, so it is never translated.
pub const ENDONYM_EN: &str = "English";
/// French endonym.
pub const ENDONYM_FR: &str = "Français";

#[cfg(test)]
mod tests {
    use super::*;
    use crate::l10n::{Catalog, En, Fr};

    /// These never-translate tokens are the AUTHORITATIVE spellings, so the
    /// catalogs must agree with them. `client_status` consumes `MIDDLE_DOT`
    /// directly; `app_name` keeps a string literal (the text-based catalog gate
    /// requires one), so this test is what makes `BRAND` load-bearing — a brand or
    /// shell-token change that updates only one side fails here.
    #[test]
    fn the_catalogs_agree_with_the_never_translate_tokens() {
        assert_eq!(En.app_name(), BRAND, "EN brand must be the canonical token");
        assert_eq!(Fr.app_name(), BRAND, "FR brand must be the canonical token");
        for status in [En.client_status("Ready"), Fr.client_status("Ready")] {
            assert!(
                status.contains(MIDDLE_DOT),
                "the status line must use the canonical middle dot: {status}"
            );
        }
    }
}
